code/modules/mob/new_player/new_player.dm

```diff
@@ -93,124 +102,105 @@
+			ready = tready
+			if(ready)
+				GLOB.ready_players++
+			else
+				GLOB.ready_players--
+		//if it's post initialisation and they're trying to observe we do the needful
+		if(!SSticker.current_state < GAME_STATE_PREGAME && tready == PLAYER_READY_TO_OBSERVE)
+			ready = tready
+			make_me_an_observer()
+			return
+
+	if(href_list["refresh"])
+		src << browse(null, "window=playersetup") //closes the player setup window
+		new_player_panel()
+
+	if(href_list["late_join"])
+		if(!SSticker || !SSticker.IsRoundInProgress())
+			to_chat(usr, "<span class='danger'>The round is either not ready, or has already finished...</span>")
+			return
+
+		if(href_list["late_join"] == "override")
 			LateChoices()
-
-
-		if("late_join_xeno")
-			if(!SSticker?.mode || SSticker.current_state != GAME_STATE_PLAYING)
-				to_chat(src, "<span class='warning'>The round is either not ready, or has already finished.</span>")
+			return
+
+		//if(SSticker.queued_players.len || (relevant_cap && living_player_count() >= relevant_cap && !(ckey(key) in GLOB.admin_datums)))
+		//	to_chat(usr, "<span class='danger'>[CONFIG_GET(string/hard_popcap_message)]</span>")
+
+		//	var/queue_position = SSticker.queued_players.Find(usr)
+		//	if(queue_position == 1)
+		//		to_chat(usr, "<span class='notice'>You are next in line to join the game. You will be notified when a slot opens up.</span>")
+		//	else if(queue_position)
+		//		to_chat(usr, "<span class='notice'>There are [queue_position-1] players in front of you in the queue to join the game.</span>")
+		//	else
+		//		SSticker.queued_players += usr
+		//		to_chat(usr, "<span class='notice'>You have been added to the queue to join the game. Your position in queue is [SSticker.queued_players.len].</span>")
+		//	return   //soon(TM)
+		LateChoices()
+
+
+	if(href_list["late_join_xeno"])
+		if(!SSticker || !SSticker.IsRoundInProgress())
+			to_chat(src, "<span class='warning'>The round is either not ready, or has already finished.</span>")
+			return
+
+		switch(alert("Would you like to try joining as a burrowed larva or as a living xenomorph?", "Select", "Burrowed Larva", "Living Xenomorph", "Cancel"))
+			if("Burrowed Larva")
+				if(SSticker.mode.check_xeno_late_join(src))
+					var/mob/living/carbon/Xenomorph/Queen/mother
+					mother = SSticker.mode.attempt_to_join_as_larva(src)
+					if(mother)
+						close_spawn_windows()
+						SSticker.mode.spawn_larva(src, mother)
+
+			if("Living Xenomorph")
+				if(SSticker.mode.check_xeno_late_join(src))
+					var/mob/new_xeno = SSticker.mode.attempt_to_join_as_xeno(src, 0)
+					if(new_xeno)
+						close_spawn_windows(new_xeno)
+						SSticker.mode.transfer_xeno(src, new_xeno)
+
+
+	if(href_list["manifest"])
+		ViewManifest()
+
+	if(href_list["SelectedJob"])
+
+		if(!GLOB.enter_allowed)
+			to_chat(usr, "<span class='notice'>There is an administrative lock on entering the game!</span>")
+			return
+		/*
+		if(SSticker.queued_players.len && !(ckey(key) in GLOB.admin_datums))
```

Comment:
Same here

Reply:
popcap system that is not implemented yet.